docs/pages/admin-guides/infrastructure-as-code/teleport-operator/secret-lookup.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: Looking up values from secrets
+description: How to store sensitive values in a Kubernetes Secret and have the operator look them up.
+---
+
+## How it works
+
+Some Teleport resources might contain sensitive values. Since Teleport 16.4, select CR fields can reference an existing
+Kubernetes secret and the operator will retrieve the value from the secret when reconciling.
+
+Currently only the GithubConnector and OIDCConnector `client_secret` field support secret lookup.
+
+## Prerequisites
+
+To follow this guide you need:
+
+- A running Teleport cluster
+- [A functional Teleport Kubernetes Operator setup](../teleport-operator.mdx#setting-up-the-operator)
+- Kubernetes rights to edit CRs and Secrets in the operator namespace
+- `kubectl` installed locally and configured for your Kubernetes cluster
+- A working GitHub or OIDC connector you want to manage with the operator
+- `tctl` and `tsh` installed and logged in the Teleport cluster
+
+## Important Considerations
+
+Even when you store sensitive values out of CRs, the CRs must still be considered as critical as
+the Kubernetes secrets themselves. Many CRs configure Teleport RBAC, someone with CR edition rights can become
+Teleport administrator and retrieve the sensitive values from Teleport.
+
+The secret lookup feature has two limitations you must take into account before configuring it:
+- for performance reasons, the secret is not watched. A secret content change is not immediately reflected on
+  the resource. To force the operator to use the new secret value, you must trigger a reconciliation by editing the CR,
+  restarting the operator, or waiting for the next full sync (every 12 hours).
+- for security reasons, the operator does allow lookup from arbitrary secrets. The secret must be annotated with
+  `resources.teleport.dev/allow-lookup-from-cr`. Possible values are `*`, or a comma-separated list of CR names.
+
+## Step 1/3. Create a Kubernetes Secret containing the sensitive value
+
+For this guide, the sensitive value we want to store is the GitHub connector client secret.
+
+Create the following `secret.yaml` manifest:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: teleport-github-connector
+  annotations:
+    # This annotation allows any CR to look up this secret
+    resources.teleport.dev/allow-lookup-from-cr: "*"
+# We use stringData instead of data for the sake of simplicity, both are OK
+stringData:
+  githubSecret: my-github-secret-value
+```
+
+If <Var name="teleport-iac" /> is your Teleport Kubernetes Operator namespace, apply the manifest using `kubectl`:
+
+```code
+$ kubectl apply -n <Var name="teleport-iac" /> -f secret.yaml
+secret/teleport-github-connector created
+```
+
+## Step 2/3. Create a CR referencing the secret
```

Comment:
```suggestion
## Step 2/3. Create a custom resource referencing the secret
```
Even if this abbreviation is common knowledge for Kubernetes users, I think it makes sense to be explicit.